src/callback_tracking.jl

```diff
@@ -297,13 +298,13 @@ function _setup_reverse_callbacks(cb::Union{ContinuousCallback, DiscreteCallback
                                                      integrator.sol.prob)
                 #vjp with Jacobin given by dw/dp before event and vector given by grad
                 vecjacobian!(dgrad, integrator.p, grad, y, integrator.t, fakeSp;
-                             dgrad = nothing, dy = nothing)
+                             dgrad = nothing, dy = nothing, recompile_tape = true)
```

Comment:
I still don't quite get why this is necessary. What branch is causing an issue here?

Reply:
it's needed because we compute vjps in the correction with respect to different functions (the affect function instead of the ODE function). So the tape precompiled for the ODE function will not work for instance when called from
https://github.com/SciML/SciMLSensitivity.jl/blob/68200c5e4ca60b3cbd2f1fd3ed76c9a380f71942/src/callback_tracking.jl#L299
for the function
https://github.com/SciML/SciMLSensitivity.jl/blob/68200c5e4ca60b3cbd2f1fd3ed76c9a380f71942/src/callback_tracking.jl#L284

Reply:
Can we just precompile a few tapes? One for each posneg, for each affect? Usually there's only a handful (vast majority one, almost never more than two) callbacks, so this isn't a huge amount.

Reply:
Sounds like a good idea. It's a little more complicated to implement because I think we have to pull the function construction/precompilation of the tape out of the affect function. I will give it a try.